README.md

```diff
@@ -124,6 +125,47 @@ Key:
 * `=` Maintenance is manual, only severe security bugs will be patched.
 * `-` Deprecated; please upgrade.
 
+#### Dependency Matrix
+Since there are multiple libraries that client-go is either required to, or can work
+with, this is a list of all the libraries and versions that work with each other.
+
+If you are using [Glide](https://glide.sh) or [godep](https://github.com/tools/godep) 
+to manage your Go dependencies, this will manage the non-optional libraries automatically.
+
+There are two ways of lining up dependencies - via the repository version *tags* 
+or via the repository *branches*.
+
+Do not mix dependencies with tags and branches between repositories, it is not supported.
+
+##### Tag Version Dependencies
+
+Using the git tag versions is the preferred way of managing dependencies for newer 
+releases of client-go
+
+| client-go    | [apimachinery](https://github.com/kubernetes/apimachinery) | [apiextensions-apiserver](https://github.com/kubernetes/apiextensions-apiserver)<sup>*</sup> | [code-generator](https://github.com/kubernetes/code-generator)<sup>+</sup> |
+| ----------------------------------| -----------------------------------------------------------|----------------------------------------------------------------------------------------------|----------------------------------------------------------------------------|
+| tag: v5.0.x  | tag: kubernetes-1.8.x  | tag: kubernetes-1.8.x | tag: kubernetes-1.8.x |
+| tag: v6.0.x  | tag: kubernetes-1.9.x  | tag: kubernetes-1.9.x | tag: kubernetes-1.9.x |
+
+Note that `x` versions need to align. For example tag v.5.0.1 in client-go, will be
+supported with tag kubernetes-1.8.1 in apimachinery and other repositories
+
+##### Branch Versions Dependencies
+
+For versions of client-go based on Kubernetes <= 1.7, the HEAD of each release 
```

Comment:
this is also the case for 1.8+. Those branch HEADs always match and are smoke tested by our scripts, plus they are CI tested in Kubernetes itself. They don't belong to an official client-go release though. So technically we do not support them. I.e. they are just like the master branch or the release branches of kube itself.

Reply:
So that makes an interesting problem for people who may need go-client <= kubernetes-1.7 - there is no way to get a supported tag / branch combo between client-go and api-machinery or other repositories, since you are telling me that release branches are not supported, and it's not supported to have a tag release + a branch release.

(This is because apimachinery, etc don't have version tags <= 1.7 on them)

Maybe they should have those tags?

What should we tell users who may need to use that specific version?

Reply:
> Maybe they should have those tags?

We could do that. There was an issue due to too few changes in the 1.7 branches which broke the "new" godep algorithm. So I did not continue with the 1.7 branch (1.8 was around the corner and hi prio).

> What should we tell users who may need to use that specific version?

As @lavalamp wrote elsewhere: the Godeps.json has valid sha1s which are proven to work. Godep and glide will pick them up to resolve the versions correctly. 

Dep won't do that, so dep users have not much help – and as you say no tags for 1.7 either. Compare @ncdc's blog https://medium.com/@andy.goldstein/upgrading-kubernetes-client-go-from-v4-to-v5-bbd5025fe381. Andy uses the branch approach there plus the semver tag. This is formally unsupported.

Reply:
Okay, so should I say that "for <= 1.7 you will need to use Glide or godep to get a supported package set of libraries or look directly at the Godep.json" ?

And then change the branch dependency matrix to say that it is tested, but isn't supported (or drop it altogether - really, I just want to make sure it's clear what _is_ supported - I'm leaning towards dropping it, it may make things cleaner and easier to understand)

WDYT?

Reply:
> Okay, so should I say that "for <= 1.7 you will need to use Glide or godep to get a supported package set of libraries or look directly at the Godep.json" ?

Sounds good.

> And then change the branch dependency matrix to say that it is tested, but isn't supported (or drop it altogether - really, I just want to make sure it's clear what is supported - I'm leaning towards dropping it, it may make things cleaner and easier to understand)

I would just drop it.

Reply:
SGTM. Will make the changes and ping when ready.